app/models/account/payment_details.rb

```diff
@@ -37,7 +37,7 @@ def create_or_build_payment_detail
       read_only_transaction = ActiveRecord::Base.connection.read_only_transaction?
       any_payment_attributes = payment_detail_attributes.any? { |_, value| value.present? }
 
-      if persisted? && any_payment_attributes && !read_only_transaction
+      if persisted? && !will_be_deleted? && any_payment_attributes && !read_only_transaction
```

Comment:
That are too many conditions (4) already ...
Maybe we should rethink of this later because it is more and more hack into hack when bugs arrive
Could you addd a JIRA about it? Like rethinking the account deletion issue

Reply:
Right :+1: